### ISSUE TITLE
Create an open street map location finder instead of the old google maps one

### ISSUE THREAD

**Issue by shai-almog** [codenameone/CodenameOne#3173]
The location simulator has commented out code that supports google maps here: 
https://github.com/codenameone/CodenameOne/blob/master/Ports/JavaSE/src/com/codename1/impl/javase/LocationSimulation.java

If you look at the history of the file this was working well and was even contributed by 3rd party https://github.com/codenameone/CodenameOne/commits?author=PiotrZub

The reason the code was commented out is that Google Maps now requires a credit card to work and we can't distribute our map key as part of an open source project. The solution of proxy won't work either since this can be sniffed. Instead this should use another map API like Open Street Map where a credit card isn't required. 

This should just require updating the HTML to that new map API and updating the binding. Ideally we'll also update this to use CEF instead of FX but I'll leave that as a separate task.

**Comment by Double-A-92** [comment]
> There are maven build instructions in the readme. Basically cd maven mvn install
> 
> Use jdk8 or jdk11
> 
> Then create a test project using start.codenameone.com
> 
> And change the cn1.version and cn1.plugin.version properties to 8.0-SNAPSHOT

Hi, I finally got it working, but I had to do little changes all over the maven configurations. I committed my changes as https://github.com/codenameone/CodenameOne/pull/3509 , maybe they might be useful to someone. 

Could you maybe give it a look to check that I didn't completely ruin something? Thank you.

**Comment by shannah** [comment]
Thanks for the PR.  I've added some comments there and requested a couple of changes.

**Comment by Double-A-92** [comment]
Unfortunately the old commented out code doesn't work anymore because of missing javafx dependencies. I read [here](https://www.codenameone.com/blog/moving-away-from-fx.html) that it has been replaced by JCFE.

Are there any examples of the new way to create a "web view" panel somewhere in the code? The old code basically created a website which loaded Google Maps, and then just displayed that inside JFXPanel.

**Comment by shai-almog** [comment]
See the code here: https://github.com/codenameone/CodenameOne/tree/master/Ports/JavaSE/src/com/codename1/impl/javase/cef It's mostly @shannah who worked on it so  I can't point at a specific block of code

**Comment by Ranzeb** [comment]
I am new to open-source contributions and I want to contribute.  Can I contribute?

**Comment by Omkar2509** [comment]
Hello, I'm new to open source contribution. I want to get started with solving some sort of issues. Can you suggest me any other good first issue or should I work on this same issue only?

**Comment by shai-almog** [comment]
Sure. You can pick any issue you want. If it's marked as [Good first issue](https://github.com/codenameone/CodenameOne/labels/good%20first%20issue) it usually means we reviewed the issue and its:

* Relatively simple
* Isolated - so you don't need to know "everything" in order to work on it

You can also look through the issue database for things that spark your interest and ask about them.

**Comment by colinldbd** [comment]
Hi, I'm a new contributor and learning how to contribute to Open Source Project, can you assign this issue to me so that I can give it a try to solve this issue?

I'll follow the contribution guide carefully.

**Comment by shai-almog** [comment]
@anniepauline yes. Sorry @colinldbd I missed your comment there, I'm assuming you moved on to something else.

**Comment by Jenish-1235** [comment]
I am beginner to open source contribution and would like to pick up this issue. Could you let me know , can I pick up this issue ?

**Comment by shai-almog** [comment]
@Jenish-1235 yes

**Comment by Jenish-1235** [comment]
I tried installing codenameone using maven and ant both of them. I facing an issue with the installation process. Probably due to my java version. I tried uninstalling my current java version and installing java 8 too. Still it is showing the same error. Attaching screenshot of the error. Could you help me to resolve it. 
![image](https://github.com/user-attachments/assets/1886019b-a692-4c89-9dc0-2d27ec86e314)

**Comment by SamC1832js** [comment]
I commited my changes in [#3863](https://github.com/codenameone/CodenameOne/pull/3863), I replaced Google Maps with OpenStreetMap for the LocationSimulation, and tested in bare bone java app using Codename One initializr. 

The latitude and longitude in the form are dynamically updated when the map position changes.

I found a known issue: When LocationSimulation is closed and reopened, the map does not display. The current workaround is to keep LocationSimulation open throughout the development process.

**Comment by shai-almog** [comment]
Fixed in https://github.com/codenameone/CodenameOne/pull/3863 by @SamC1832js 

@all-contributors please add @SamC1832js  for code
